doc/elements-tx-format.md

```diff
@@ -0,0 +1,570 @@
+
+#### TxOut
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset | Yes | 1 or 33 bytes | `ConfidentialAsset` | |  |
+| Amount | Yes | 1 or 9 or 33 bytes | `ConfidentialAmount` | | |
+| Nonce | Yes | 1 or 33 bytes | `ConfidentialNonce` | | |
+| ScriptPubKey Length | Yes | Varies | `VarInt` | | | |
+| ScriptPubKey | If ScriptPubKey Length is non-zero | Varies | `hex` | Big endian | |
+
+Serialized format: `[Asset][Amount][Nonce][ScriptPubKey Length][ScriptPubKey]`
+
+#### Witness
+
+As noted above, the witness is only present for segwit transactions.
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Input Witnesses| Yes | Varies | `Vector<InputWitness>` | | There is exactly one input witness for each input in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of inputs. |
+| Output Witnesses| Yes | Varies | `Vector<OutputWitness>` | | There is exactly one output witness for each output in the transaction.<br><br>This number is not explicitly included in the witness — it is implied by the number of outputs. |
+
+Serialized format: `[Input Witnesses][Output Witnesses]`
+
+#### InputWitness
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Issuance Amount Range Proof | Yes | Varies | `Proof` | | `0x00` → null. |
+| Inflation Keys Range Proof | Yes | Varies | `Proof` | | `0x00` → null. |
+| Script Witness | Yes | Varies | `Vec<hex>` | | The vector represents the witness stack.<br>Can be empty (length of 0). |
+| Pegin Witness | Yes | Varies | `Vec<hex>` | | The vector represents the witness stack.<br>Can be empty (length of 0). |
+
+Serialized format:
+`[Issuance Amount Range Proof][Inflation Keys Range Proof][Script Witness][Pegin Witness]`
+
+The range proofs must be empty if their asociated amount (issuance / inflation keys) is explicit.
+
+#### OutputWitness
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Surjection Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+| Range Proof | Yes | Varies | `Proof` | | Will be null (`0x00`) if corresponding output is not confidential. |
+
+Serialized format: `[Surjection Proof][Range Proof]`
+
+The Range Proof must be empty if the output’s amount is explicit. 
+
+The Surjectionproof must be empty if the output’s assetId is explicit.
+
+#### AssetIssuance
+
+|Field|Required|Size|Data Type|Encoding|Notes| 
+|-----|--------|-----|---------|--------|-----|
+| Asset Blinding Nonce | Yes | 32 bytes | `hex` | | Zero for a new asset issuance; otherwise a blinding factor for the input. |
+| Asset Entropy | Yes | 32 bytes | `hex` | | Freeform entropy field, no consensus-defined meaning, but is used as additional entropy to the asset tag calculation. |
```

Comment:
This is only freeform for new issuances.  For reissuances it is required to be the asset's entropy value.

Reply:
Added.